src/components/carousel/ThumbnailPagination.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {
+  FC,
+  LegacyRef,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { EmblaCarouselType } from 'embla-carousel-react';
+
+import { useMultiStyleConfig } from '@chakra-ui/react';
+
+import ThumbnailNavigationButton from './ThumbnailNavigationButton';
+import Thumbnail from './Thumbnail';
+import Flex from '../flex';
+import Box from '../box';
+
+interface Props {
+  currentSlideIndex: number;
+  thumbnails: ReactNode[];
+  mainCarousel?: EmblaCarouselType;
+  paginationCarouselRef?: LegacyRef<HTMLDivElement>;
+  paginationCarousel?: EmblaCarouselType;
+}
+
+const ThumbnailPagination: FC<Props> = ({
+  currentSlideIndex,
+  thumbnails,
+  mainCarousel,
+  paginationCarouselRef,
+  paginationCarousel,
+}) => {
+  const [thumbnailScrollProgress, setThumbnailScrollProgress] = useState(0);
+  const [showPaginationButtons, setShowPaginationButtons] = useState(true);
+  const { pagination } = useMultiStyleConfig('Carousel', {
+    variant: 'fullScreen',
+  });
+
+  const scrollPrev = useCallback(
```

Comment:
I copied them from the official embla-carousel documentation: https://codesandbox.io/s/embla-carousel-arrows-dots-react-z5fbs?file=/src/js/EmblaCarousel.js:612-690. Perhaps it's fine to remove it. Let me check a bit

Reply:
Haven't found any issues with it but if it's ok I'd like to follow the documentation just to be safe. The dependency should be stable and there are hundreds of rerenders because of the scroll progress.